pkg/provisioner/ironic/ironic.go

```diff
@@ -29,15 +30,32 @@ var log = logf.Log.WithName("baremetalhost_ironic")
 var deprovisionRequeueDelay = time.Second * 10
 var provisionRequeueDelay = time.Second * 10
 var powerRequeueDelay = time.Second * 10
+var deployKernelURL string
+var deployRamdiskURL string
+var ironicEndpoint string
 
 const (
-	ironicEndpoint = "http://localhost:6385/v1/"
 	// See nodes.Node.PowerState for details
 	powerOn   = "power on"
 	powerOff  = "power off"
 	powerNone = "None"
 )
 
+func init() {
+	deployKernelURL = os.Getenv("DEPLOY_KERNEL_URL")
+	if deployKernelURL == "" {
+		panic("No DEPLOY_KERNEL_URL variable set")
```

Comment:
nit: I'm not a fan of `panic` unless stuff really hits the fan. It outputs a stacktrace along with the message, which isn't helpful to an end user. They just need to know that they should set the environment variable. I think it would be better to log a message and then `os.Exit(1)`.

Reply:
Fixed